packages/flutter_tools/lib/src/debug_adapters/flutter_adapter.dart

```diff
@@ -431,29 +421,28 @@ class FlutterDebugAdapter extends DartDebugAdapter<FlutterLaunchRequestArguments
         //   adapter once rolled into Flutter.
         sendEvent(
           RawEventBody(<String, Object?>{
-            'vmServiceUri': serviceUri.toString(),
+            'vmServiceUri': vmServiceUri.toString(),
           }),
           eventType: 'dart.debuggerUris',
         );
       }
-    }
   }
 
   /// Handles the app.start event from Flutter.
   void _handleAppStart(Map<String, Object?> params) {
-    _appId = params['appId'] as String?;
-    assert(_appId != null);
+    appId = params['appId'] as String?;
+    assert(appId != null);
```

Comment:
should we throw a `StateError` in this case?

Reply:
Yeah, throwing makes sense. Although I used `DebugAdapterException` instead of `StateError` since it's not really related to the state of this object, if we get a `null` here it means `flutter run --machine` has emitted an invalid `app.start` event (without a valid `appId`).